packages/discord.js/typings/index.d.ts

```diff
@@ -5190,6 +5212,8 @@ export interface GuildMembersChunk {
   nonce: string | undefined;
 }
 
+type NonPartialGroupDMChannel<Structute> = Structute & { channel: Exclude<TextBasedChannel, PartialGroupDMChannel> };
```

Comment:
This currently doesn't respect the `InGuild` type parameter on Message and will add `DMChannel` to a `Message<true>`.

```suggestion
type NonPartialGroupDMChannel<Structure extends {channel: Channel}> = Structure & {
  channel: Exclude<Structure['channel'], PartialGroupDMChannel>;
};
```

Reply:
It does respect